packages/xo-web/src/xo-app/vm-import/esxi/esxi-import.js

```diff
@@ -0,0 +1,262 @@
+import _, { messages } from 'intl'
+import ActionButton from 'action-button'
+import Button from 'button'
+import decorate from 'apply-decorators'
+import React from 'react'
+import { connectStore } from 'utils'
+import { createGetObjectsOfType } from 'selectors'
+import { esxiConnect, importVmFromEsxi, isSrWritable } from 'xo'
+import { find, isEmpty, map, pick } from 'lodash'
+import { injectIntl } from 'react-intl'
+import { injectState, provideState } from 'reaclette'
+import { Input } from 'debounce-input-decorator'
+import { InputCol, LabelCol, Row } from 'form-grid'
+import { linkState } from 'reaclette-utils'
+import { Password, Select } from 'form'
+import { SelectNetwork, SelectPool, SelectSr } from 'select-objects'
+
+import VmData from './vm-data'
+
+const getInitialState = () => ({
+  hasCertificate: true,
+  hostIp: '',
+  isConnected: false,
+  network: undefined,
+  password: '',
+  pool: undefined,
+  sr: undefined,
+  user: '',
+  vm: undefined,
+  vmData: undefined,
+  vmsById: undefined,
+})
+
+const EsxiImport = decorate([
+  connectStore({
+    hostsById: createGetObjectsOfType('host'),
+    pifsById: createGetObjectsOfType('PIF'),
+  }),
+  provideState({
+    initialState: getInitialState,
+    effects: {
+      importVm:
+        () =>
+        async ({ defaultNetwork, hasCertificate, hostIp, network, password, sr, user, vm }) => {
+          await importVmFromEsxi({
+            host: hostIp,
+            network: network?.id ?? defaultNetwork,
+            password,
+            sr,
+            sslVerify: hasCertificate,
+            user,
+            vm: vm.value,
+          })
+        },
+      connect:
+        () =>
+        async ({ hostIp, hasCertificate, password, user }) => {
+          const vms = await esxiConnect(hostIp, user, password, hasCertificate)
+          return { isConnected: true, vmsById: vms.reduce((vms, vm) => ({ ...vms, [vm.id]: vm }), {}) }
+        },
+      linkState,
+      onChangeVm: (_, vm) => state => ({ vm, vmData: state.vmsById[vm.value] }),
```

Comment:
Can't we just pass `vms` to the Select so that we don't need to have `vm` *and* `vmData`?